CCPA/Data Deletion Request Handling.md

```diff
@@ -0,0 +1,329 @@
+4. Vendors perform deletion asynchronously, in accordance with their technology and operational practices. 
+
+
+### How should publishers load vendor scripts?
+
+Vendors need access to the publisher domain to get a user identifier to associate with a delete request. In order for vendors to do so, the vendor scripts described in this specification must be loaded _directly _on the publisher domain normally used for collecting data and serving ads and not on a separate or generic domain. For safety, the vendor scripts can be loaded in an iframe, provided the following conditions are met:
+*   The iframe is hosted on the publisher domain
+*   If the iframe sandbox parameter is provided, the following restrictions are lifted:
+    *   `allow-scripts`
+    *   `allow-storage-access-by-user-activation`
+
+
+## registerDeletion
+
+This command registers a vendor-specific callback function at the API. When the publisher loads the vendor deletion script, those scripts must execute `registerDeletion`. 
+
+This command is an update to the existing `__uspapi` function with the following syntax: 
+
+```
+__usapi("registerDeletion", version, performDeletionFunction)
+```
+
+Upon execution, the vendor script is registered to receive the signal sent by the call to `performDeletion`.
+
+
+## performDeletion
+
+The publisher, or its CMP where applicable, calls this command based on some user delete action to initiate the deletion process. The command communicates to vendors that a specified user has requested that their personal data be deleted by calling the `performDeletionFunction` registered during the call to `registerDeletion`. 
+
+This command is an update to the existing `__uspapi` function with the following syntax: 
+
+```
+__usapi("performDeletion", version, null, identifiers)
+```
+
+The callback parameter of the `__uspapi` is not used in this case and can remain null. The `identifiers` parameter is only required when handling non-web deletion requests and is further explained in the following section. 
+
+
+### Identifying Data to Delete in Non-web Environments
+
+When operating in a non-web environment, data deletion requests are handled by sending the user to a web page where they can complete the request to have their data deleted. Vendors need certain information to correctly identify the data to delete:  the platform name, the unique app identifier used in the app store, and the device identifier for that platform / store. 
+
+The `performDeletion` command includes a parameter for identifiers. Using the `identifiers` parameter, publishers can pass multiple items, each with the required fields: `platform`, `app_identifier`, and `user_identifier`. These details should be passed from the app context to the delete webpage and then along with the request from the page. Below are a few examples of that information:
```

Comment:
I don't think it is ever possible to have more than one `platform`, `app_identifier`, or `user_identifier` in a single app.

Reply:
You may need to pass the deletion for multiple apps at the same time?

Reply:
@Facens this software only runs in one App at a time